File: app/routers/report.py
from fastapi import APIRouter, Query
from app.db import get_history

router = APIRouter(prefix="/api/report", tags=["Report"])

@router.get("")
def report(
    action: str | None = Query(None),
    start_date: str | None = Query(None),
    end_date: str | None = Query(None)
):
    """
    통합 리포트
    - action: INBOUND | OUTBOUND | MOVE | None
    - 날짜는 YYYY-MM-DD
    """
    rows = get_history(
        action=action,
        start_date=start_date,
        end_date=end_date
    )
    return rows

File: app/static/js/report.js
const actionMap = {
  "INBOUND": "입고",
  "OUTBOUND": "출고",
  "MOVE": "이동"
};

async function loadReport() {
  const action = document.getElementById("action").value;
  const start = document.getElementById("start").value;
  const end = document.getElementById("end").value;

  const params = new URLSearchParams();
  if (action) params.append("action", action);
  if (start) params.append("start_date", start);
  if (end) params.append("end_date", end);

  const res = await fetch("/api/report?" + params.toString());
  const data = await res.json();

  const tbody = document.getElementById("reportBody");
  tbody.innerHTML = "";

  data.forEach(r => {
    const tr = document.createElement("tr");
    tr.innerHTML = `
      <td>${actionMap[r.action] || r.action}</td>
      <td>${r.item_code}</td>
      <td>${r.lot || ""}</td>
      <td>${r.location_from || ""}</td>
      <td>${r.location_to || ""}</td>
      <td style="text-align:right">${r.quantity}</td>
      <td>${r.created_at.replace("T", " ")}</td>
    `;
    tbody.appendChild(tr);
  });
}
